mempool/clist_mempool.go

```diff
@@ -524,12 +534,18 @@ func (mem *CListMempool) ReapMaxBytesMaxGas(maxBytes, maxGas int64) types.Txs {
 
 // Safe for concurrent use by multiple goroutines.
 func (mem *CListMempool) ReapMaxTxs(max int) types.Txs {
-	mem.updateMtx.RLock()
-	defer mem.updateMtx.RUnlock()
-
+	memTxsLen := mem.txs.Len() // concurrency safe
 	if max < 0 {
-		max = mem.txs.Len()
+		max = memTxsLen
 	}
+	canExecuteConcurrentlyWithRecheck := max < memTxsLen
```

Comment:
if `max = memTxsLen` on line 539, then this condition will be false, correct?

Reply:
Yes! (That is the intention, as we would have to wait until recheck is done by definition)